### PR TITLE
Supra docs

### PR DESCRIPTION
**Pull request by symbolpunk**
Pulled from https://github.com/OffchainLabs/arbitrum-docs/pull/480 to `yarn format`, apply frontmatter, etc

### CODE REVIEW

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+Supra is a novel, high-throughput Oracle & IntraLayer: A vertically integrated toolkit of cross-chain solutions (data oracles, asset bridges, automation network, and more) that interlink all blockchains, public (L1s and L2s) or private (enterprises).
+
+Integrating with Supra's price feeds is quick and easy. Supra currently supports several Solidity/EVM-based networks, like Arbitrum, and non-EVM networks like Sui, Aptos.
+
+To see all of the networks Supra is on, please visit [Supras' Networks](https://supraoracles.com/docs/price-feeds/networks)!
+
+To get started, you will want to visit [Supras' docs site](https://supraoracles.com/docs/price-feeds/) and review the documentation or continue to follow this guide for a quick start.
+
+### Step 1: Create The S-Value Interface
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:23]:
```suggestion
### Step 1: Create The S-Value interface
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    function getSvalue(uint64 _pairIndex) external view returns (bytes32, bool);
+
+    function getSvalues(uint64[] memory _pairIndexes) external view returns (bytes32[] memory, bool[] memory);
+
+}
+```
+
+This creates the interface that you will later apply in order to fetch a price from SupraOracles.
+
+### Step 2: Configure The S-Value Feed Address
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:39]:
```suggestion
### Step 2: Configure The S-Value feed address
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+To fetch the S-Value from a SupraOracles smart contract, you must first find the S-Value Feed Address for the chain of your choice.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:41]:
```suggestion
To fetch the S-Value from a SupraOracles smart contract, you must first find the S-Value feed address for the chain of your choice.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+For Arbitrum, the address is: 0x8a358F391d93f7558D5F5E61BDf533e2cc3Cf7a3
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:43]:
```suggestion
For Arbitrum, the address is [0x8a358F391d93f7558D5F5E61BDf533e2cc3Cf7a3](https://arbiscan.io/address/0x8a358f391d93f7558d5f5e61bdf533e2cc3cf7a3)
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+When you have the proper address, create an instance of the S-Value Feed using the interface we previously defined for Arbitrum:
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:45]:
```suggestion
When you have the proper address, create an instance of the S-Value feed using the interface we previously defined for Arbitrum:
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    function bytesToUint256(bytes memory _bs) internal pure returns (uint256 value) {
+        require(_bs.length == 32, "bytes length is not 32.");
+        assembly {
+            value := mload(add(_bs, 0x20))
+        }
+    }
+```
+
+### Step 4: Get The S-Value Crypto Price
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:80]:
```suggestion
### Step 4: Get the S-Value crypto price
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+Now you can simply access the S-Value Crypto Price of our supported market pairs. In this step, we'll get the price of single or multiple trading pairs in our smart contract.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:82]:
```suggestion
Now you can simply access the S-Value crypto price of our supported market pairs. In this step, we'll get the price of single or multiple trading pairs in our smart contract.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+            uint256[4] memory decoded = unpack(val[i]);
+            decodedArray[i] = decoded;
+        }
+
+        return decodedArray;
+}
+```
+
+Tada! You now have a method in your Smart Contract that you can call at any time to retrieve the price of any crypto pair!
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:110]:
```suggestion
Tada! You now have a method in your smart contract that you can call at any time to retrieve the price of any crypto pair!
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+## Going Further with Supra
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle.mdx:112]:
```suggestion
## Going further with Supra
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+Latest version of Supra VRF requires a customer controlled wallet address to act as the main reference for access permissions and call back(response) transaction gas fee payments. Therefore, users planning to consume Supra VRF should get in touch with our team to get your wallet registered with Supra.
+
+Once your wallet is registered by the Supra team, you could use it to whitelist any number of VRF requester smart contracts and pre-pay/top up the deposit balance maintained with Supra in order to pay for the gas fees of callback(response) transactions.
+
+You will be interacting with two main contracts:
+
+- **Supra Deposit Contract:** To whitelist smart contracts under the registered wallet address, pre-pay/top up the callback gas fee deposit maintained with Supra.
+- **Supra Router Contract:** To request and receive random numbers.
+
+## Step 1: Create the Supra Router Contract Interface​
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:32]:
```suggestion
## Step 1: Create the Supra router contract interface​
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+
+Add the following code to the requester contract i.e, the contract which uses VRF as a service. You can also add the code in a separate Interface and inherit the interface in the requester contract.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:34]:
```suggestion
Add the following code to the requester contract i.e, the contract which uses VRF as a service. You can also add the code in a separate interface and inherit the interface in the requester contract.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+
+```
+interface ISupraRouterContract {
+	function generateRequest(string memory _functionSig, uint8 _rngCount, uint256 _numConfirmations, uint256 _clientSeed, address _clientWalletAddress) external returns(uint256);
+	function generateRequest(string memory _functionSig, uint8 _rngCount, uint256 _numConfirmations, address _clientWalletAddress) external returns(uint256);
+}
+```
+
+This interface will help the requester contract interact with the Supra Router contract and through which the requester contract can use the VRF service.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:43]:
```suggestion
This interface will help the requester contract interact with the Supra router contract and through which the requester contract can use the VRF service.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+
+## Step 2: Configure the Supra Router Contract Address​
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:45]:
```suggestion
## Step 2: Configure the Supra router contract address​
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+
+Contracts that need random numbers should utilize the Supra Router Contract. In order to do that, they need to create an interface and bind it to the on-chain address of the Supra router contract.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:47]:
```suggestion
Contracts that need random numbers should utilize the Supra router contract. In order to do that, they need to create an interface and bind it to the on-chain address of the Supra router contract.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+contract ExampleContract {
+    ISupraRouter internal supraRouter;
+
+    constructor(address routerAddress) {
+        supraRouter = ISupraRouter(0x7d86fbfc0701d0bf273fd550eb65be1002ed304e);
+    }
+}
+```
+
+## Step 3: Use the VRF service and request a Random Number​
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:59]:
```suggestion
## Step 3: Use the VRF service and request a random number​
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+Inside the callback function where the requester contract wants the random number (in this example the callback function is exampleCallback), the requester contract will have to add the validation such that only the Supra router contract can call the function. The validation is necessary to protect against malicious contracts/users executing the callback with fake data. For example, if the callback function is pickWinner in the requester contract, the snippet can be as follows.
+
+```
+function exampleCallback(uint256 _nonce ,uint256[] _rngList) external {
+    require(msg.sender == address(SupraRouter));
+    // Following the required logic of the function
+ }
+```
+
+## Step 5 : Whitelist your requester contract with Supra Deposit Contract and deposit funds​
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:100]:
```suggestion
## Step 5 : Whitelist your requester contract with Supra deposit contract and deposit funds​
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+	function checkMinBalance(address _clientAddress) external view returns(uint256);
+	function countTotalWhitelistedContractByClient(address _clientAddress) external view returns (uint256);
+	function getSubscriptionInfoByClient(address _clientAddress) external view returns (uint256, uint256, bool);
+	function isMinimumBalanceReached(address _clientAddress) external view returns (bool);
+	function listAllWhitelistedContractByClient(address _clientAddress) external view returns (address[] memory);
+
+}
+```
+
+Navigate to the “Navigate & run Transactions” tab in remix, and Paste the Deposit Contract Address into the text box besides the “At Address” button & Press the At Address button. You will find the instance for the Deposit Contract created using which a user can interact and use the features provided by the Deposit Contract.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:126]:
```suggestion
Navigate to the “Navigate & run Transactions” tab in remix, and paste the deposit contract address into the text box besides the “At Address” button & press the at address button. You will find the instance for the deposit contract created using which a user can interact and use the features provided by the deposit contract.
```

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 18]
```diff
@@ -0,0 +1,185 @@
...
+“addContracttoWhitelist(address)” - The whitelisted users will have to whitelist their contract which they will be using to request for random numbers. The parameter this function takes is the User’s contract address. This function will be called after the user deploys the requester contract post development and makes it ready for interacting with the Supra Contracts.
+“depositFundClient()” - is another mandatory function for a user to use once, before the user starts requesting from that contract. This is a function which will deposit funds in the deposit contract from the users for the response/callback transaction. The funds for a specific user should remain higher than the minimum amount set by the Supra( 0.1 Eth for Arbitrum testnet) for the new request transactions to be accepted.  
+Basically the gist here is that the user will have to interact with the Deposit contract and add funds for their accounts, which will be utilized for the response transaction gas fee. There will be a script from Supra which will be monitoring the funds and will alert the user if a refill is required.
+
+## Example Implementation
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:132]:
```suggestion
## Example implementation
```

**Review by mahsamoosavi — CHANGES_REQUESTED** [review]

File: arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx [hunk 19]
```diff
@@ -0,0 +1,185 @@
...
+         rngForUser[result[nonce]][i] = rngList[i] % 100;
+      }
+   }
+   function viewUserName(string memory username) external view returns (uint256[] memory) {
+      return rngForUser[username];
+   }
+   }
+```
+
+## Going Further with Supra
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/third-party-docs/Supra/use-supras-vrf.mdx:168]:
```suggestion
## Going further with Supra
```

**Review by mahsamoosavi — DISMISSED** [review]

File: arbitrum-docs/devs-how-tos/how-to-use-oracles.mdx [hunk 20]
```diff
@@ -249,6 +249,8 @@ contract ARBPriceConsumer is UsingTellor {
 
 You can adapt this contract to your needs. Just remember to use the ticker of the assets you want to request the price for, and to **deploy your contract to the appropriate network, with the address of the oracle contract in that network**. Remember we have a [Quickstart](../for-devs/quickstart-solidity-hardhat) available that goes through the process of compiling and deploying a contract.
 
-### More examples
+### See also
 
-Refer to [Tellor’s documentation](https://docs.tellor.io/) for more examples of querying price feeds plus other data feeds available.
+- [Tellor’s documentation](https://docs.tellor.io/) demonstrates how to query price feeds and other data feeds.
+- [How to use Supra's price feed oracle](/for-devs/third-party-docs/Supra/use-supras-price-feed-oracle)
```
Comment (TucksonDev) [inline arbitrum-docs/devs-how-tos/how-to-use-oracles.mdx:255]: This is at the same level as Tellor's section. I noticed the original contributor placed the links there, but it looks like they should go in a different section, right?
Maybe this "See also" section can be an h2 (##) ?

**Review by TucksonDev — APPROVED** [review]
This is amazing @symbolpunk ! 🚀 
Approving to unblock, but take a look at my previous comment about the links to Supra's docs in "How to use Oracles" 🙏 
Thanks!

File: arbitrum-docs/devs-how-tos/how-to-use-oracles.mdx (hunk 20 above)
Comment (symbolpunk) [reply arbitrum-docs/devs-how-tos/how-to-use-oracles.mdx:255]: Acknowledging receipt - will follow up!

**Pull request merged**